src/math/calculation.js

```diff
@@ -829,4 +829,46 @@ function hypot(x, y, z) {
   return Math.sqrt(sum) * max;
 }
 
+/**
+ * Calculates the fractional part of a number.
+ *
+ * @method fract
+ * @param {Number} num Number whose fractional part needs to be found out
+ * @returns {Number} fractional part of x, i.e, {x}
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(windowWidth, windowHeight);
+ *   fill(0);
+ *   text(7345.73472742, 0, 50);
+ *   text(fract(7345.73472742), 0, 100);
+ *   text(1.4215e-15, 150, 50);
+ *   text(fract(1.4215e-15), 150, 100);
+ * }
+ * </code>
+ * </div>
+ * @alt
+ * 2 rows of numbers, the first row having 8 numbers and the second having the fractional parts of those numbers.
+ */
+p5.prototype.fract = function(num) {
+  p5._validateParameters('fract', arguments);
+  let sign = 0;
+  if (num === undefined || Math.abs(num) === Infinity) {
```

Comment:
Can you explain a bit about why there's a check for `undefined` here? Mainly because there are several other cases to handle if you want to handle `undefined` (`null`, `NaN`, `"text string"`, etc). Any of the above values should already be caught with the FES but if you want to catch them and return them as is (the unhandled ones return 0 at the moment), you can just check here if `num` is a Number.

Reply:
Well, firstly my initial reason to handle `undefined` was that if not it showed an error (expected number of arguments), but then I thought that if by any mistake, I put a `null, NaN, "text string"` in there, I wouldn't want 0 to be returned, because the nature of `fract()` makes sure that 0 is a very important number and can cause a lot of errors in computations. And I think I forgot changing the check to checking if `num` is a number. I think that change would suffice most incorrect inputs?

Reply:
Revert back to me on this and I'll make the necessary changes as soon as you do.

Reply:
Yeah, just make the check for the number type. I don't think it is necessary to specially handle `undefined` since it would be taken care of by the number type check and it wouldn't suppress the FES message anyway (not that it's necessary).

Reply:
```javascript
function fract(num){
    let sign = 0;
    num=Number(num)
    if (typeof num != Number || Math.abs(num) === Infinity) return num;
    else if (num < 0) {
      num = -num;
      sign = 1;
    }
    if (String(num).includes('.') && !String(num).includes('e')){
      let toFract = String(num);
      toFract = Number('0' + toFract.slice(toFract.indexOf('.')))
      return Math.abs(sign-toFract);
    } else if (num<1) return Math.abs(sign-num);
    else return 0;
};

console.log(fract(7345.73472742))
console.log(fract(41251521531625123e-25))
console.log(fract("HelloWorld"))
console.log(fract("21749028.42154"))
console.log(fract(NaN))
console.log(fract())
``` 

Hey, how about something like this?

Reply:
I performed the `Number` typecasting at line 3 to take care of something like example 4.

Reply:
`typeof` returns a string so it should be `typeof num != "number"`.

Reply:
Oh, right. I'm sorry. I've updated the code accordingly.

Reply:
I'm sorry. There still seems to be something wrong with this.

Reply:
I've updated the code. I think it should work now (just had a brainfade moment).